code/mob/living/critter/sawfly.dm

```diff
@@ -0,0 +1,206 @@
+/*
+->The sawfly, by NightmareChamillian
+This file is the critter itself, and all the custom procs it needs in order to function.
+
+-For the AI, check the critter/AI folder, it should be in sawflyai.dm
+-For the grenade and controller, check code/obj/sawflymisc.dm
+*/
+/mob/living/critter/robotic/sawfly
+
+	name = "Ranodyne antipersonnel microdrone"
+	desc = "A folding antipersonnel drone, made by Ranodyne LLC. It'd be pretty cute if it wasn't trying to kill people."
+	icon = 'icons/obj/ship.dmi'//remnants of it originally being a drone
+	icon_state = "sawfly"
+	flags = TABLEPASS
+
+	var/beeptext = " "
+	var/deathtimer = 0 // for catastrophic failure on death
+	var/isnew = TRUE // for seeing whether or not they will make a new name on redeployment
+	var/sawflynames = list("A", "B", "C", "D", "E", "F", "V", "W", "X", "Y", "Z", "Alpha", "Beta", "Gamma", "Lambda", "Delta")
+	var/beeps = list('sound/machines/sawfly1.ogg','sound/machines/sawfly2.ogg','sound/machines/sawfly3.ogg') // custom noises so they cannot be mistaken for ghostdrones or borgs
+	health = 50 //this value's pretty arbitrary, since it's overridden when they get their healtholders
+	var/fliesnearby = 0 //for rolling chance to beep
+	var/friends = list()
+	misstep_chance = 40 //makes them behave more like drones, and harder to kite into a straightaway then shoot
+	var/list/dummy_params = list("icon-x" = 16, "icon-y" = 16) //for the manual attack_hand retaliation
+
+	//mob variables
+	custom_gib_handler = /proc/robogibs
+	isFlying = 1
+	can_grab = FALSE
+	can_help = FALSE
+	can_disarm = FALSE
+	use_stamina = FALSE
+	use_stunned_icon = FALSE
+	butcherable = FALSE
+	can_bleed = FALSE
+	canbegrabbed = FALSE
+	can_lie = FALSE
+	can_burn = FALSE
+	pet_text = "cuddles"
+	hand_count = 1 //stabby hands
+	setup_healths()
+		add_hh_robot(25, 1)
+		add_hh_robot_burn(25, 1)
+
+
+
+	New()
+		..()
+		if(isnew)
+			name = "Sawfly [pick(sawflynames)]-[rand(1,999)]"
+		deathtimer = rand(1, 5)
+		death_text = "[src] jutters and falls from the air, whirring to a stop."
```

Comment:
why do this in new instead of following the general pattern for mobcritters? (doing it in typedef with `%src%` in the string, which gets parsed in death handling code)